pkg/controller/submariner/submariner_controller.go

```diff
@@ -247,6 +250,24 @@ func (r *ReconcileSubmariner) Reconcile(request reconcile.Request) (reconcile.Re
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileSubmariner) retrieveDaemonSetContainerStatuses(daemonSet *appsv1.DaemonSet,
```

Comment:
Also, I'm not sure if this is something we may want to do on every single reconcile loop. We'd be grabbing lots of containers at scale. 

What if we setup an informer to keep up to date with those pods, we could collect anything we need, or trigger a reconcile loop if something goes wrong?